api/src/main/java/javax/mvc/engine/ViewEngineContext.java

```diff
@@ -74,21 +71,6 @@
     Models getModels();
 
     /**
-     * Returns HTTP request object from the Servlet container.
-     *
-     * @return HTTP request object.
-     */
-    HttpServletRequest getRequest();
-
-    /**
-     * Returns HTTP response object from the servlet container. The underlying
-     * output stream should be used to write the result of processing a view.
-     *
-     * @return HTTP response object.
-     */
-    HttpServletResponse getResponse();
-
```

Comment:
Think I once was in a situation where someone set a caching header before me and the only thing I could do is setting the header again with a nonsense value. But can't event tell you the used framework any more...

Didn't try but I think `Object` could be `List` if the same header is set twice.

Without adding the `PrintWriter` to the interface we don't have any chance to control the encoding from outside the `ViewEngine`. I'm not too happy with such a decision because I would prefer `@Produces`. I would be fine if two things are added to the spec:
* We have a configuration property where an implementer of a `ViewEngine` and a application developer can define the Content-Type and encoding and this property always wins. We must take in account that many template engines don't have a way to configure this.
* An implementation SHOULD warn if `@Produces("text/html")` is used on a controller method (no matter if with or without charset).